documentation/decisions/adr/000N-frontend-tooling.md

```diff
@@ -0,0 +1,82 @@
+# Front-end Code Quality Tools
+
+- **Status:** Proposed <!-- REQUIRED -->
+- **Last Modified:** 2023-07-13 <!-- REQUIRED -->
+- **Related Issue:** [#102](https://github.com/HHS/grants-api/issues/102) <!-- RECOMMENDED -->
+- **Deciders:** Aaron Couch, Daphne Gold, Sammy Steiner, Gina Carson, Lucas Brown, Billy Daly <!-- REQUIRED -->
+- **Tags:** ADR <!-- OPTIONAL -->
+
+## Context and Problem Statement
+
+Front-end code quality tools should facilitate and efficiently enforce linting, auto-formatting, type-checking and security concerns. They should be easily configurable to suit our specific use-cases, uniformly applying conventions while alleviating the need for individual intervention.
+
+## Decision Drivers <!-- RECOMMENDED -->
+
+- **Ease of use and configurability:** Code quality tools should be configurable to meet the specific needs of our project and enforce standards uniformly. Running the tool should be simple, and feedback should be easily implementable and understandable.
+- **Speed:** Code quality tools should be able to run in an timely manner, ideally in a pre-commit hook.
+- **Documentation and resources:** Code quality tools should have robust and helpful documentation, specifically around usage and configuration.
+- **Lean:** Selection of code quality tools should cover all use-cases with little overlap for ease of use and to minimize mental surface area required to understand the system. As in, each tool should have a specific, meaningful purpose.
+- **Enforces security:** Tooling should assist us in identifying security concerns and vulnerabilities.
+- **Well-maintained:** We have a preference towards libraries that are widely adopted and have active maintainers.
+
+## Options Considered
```

Comment:
do we want to consider something like `immutableJS` to enforce immutability? i know there are lots of pros and cons to that!

Reply:
Would this be alongside Typescript? Typescript will add immutability functionality that we can use as we desire, and I think that's sufficient for now. I'd need to do a deeper dive into immutable.js to see if it offers something additional. I can add it to the documentation so it is tracked as an option.